pkg/chart/chart.go

```diff
@@ -222,6 +245,23 @@ func (t *Testing) processCharts(action func(chart string, valuesFiles []string)
 		TestResults:    results,
 	}
 
+	// Checkout previous chart revisions and build their dependencies
+	if t.config.Upgrade {
```

Comment:
As mentioned earlier, this is also called when linting, which doesn't make sense.

Reply:
Would you prefer that we implement this in `InstallCharts` and `LintAndInstallCharts` instead, or somewhere else? The reason I haven't done this was that I wanted to avoid calling `FindChartsToBeProcessed` multiple times or adding more args to `processCharts` if possible. It is also a bit more complex to do this prep in a utility function since we'd have to pass a cleanup callback function to remove the working tree rather than just using defer.

I feel like it makes sense to have a run-once "prepare" function. Here is the current behavior as I understand it:
- If `ct lint` is run, `--upgrade` is not an option and this code branch will never be executed (ct exits with `unknown flag: --upgrade`).
- If `ct lint-and-install` is run, this code branch is only executed once.

There may also come a time when we want previous chart revisions available for more than the upgrade testing use case. For example, the git tool's `FileExistsOnBranch` and `Show` methods could both be implemented using the previous revision's worktree rather than going over the network. If/when this were the case, keeping this functionality directly in `processCharts` would make that a very minimal refactor.

If you're still not convinced then no worries, please just let me know where this should be implemented. 😄

Reply:
I commented again about this further above (https://github.com/helm/chart-testing/pull/103#discussion_r264227348), probably at the wrong place:

> Yeah, I figured going through the code again. However, `Upgrade` can be `true` if set as env var or via config file when linting. We should explicitly set it to `false` when the config is loaded in this case below this line: https://github.com/helm/chart-testing/blob/master/pkg/config/config.go#L113